$OpenBSD: patch-src_lastfm_h,v 1.1 2015/06/19 16:23:33 dcoppa Exp $

commit 20a4e486cc36e40afc202e3794772e402b1d3c0f
Author: Andrzej Rybczak <electricityispower@gmail.com>
Date:   Sat May 23 19:33:20 2015 +0200

lastfm: use sink argument and fix possible usage of m_service after
delete

--- src/lastfm.h.orig	Sat May  2 21:16:11 2015
+++ src/lastfm.h	Fri Jun 19 16:36:58 2015
@@ -51,18 +51,16 @@ struct Lastfm: Screen<NC::Scrollpad>, Tabbable
 	virtual bool isMergable() OVERRIDE { return true; }
 	
 	template <typename ServiceT>
-	void queueJob(ServiceT &&service)
+	void queueJob(ServiceT *service)
 	{
-		typedef typename std::remove_reference<ServiceT>::type ServiceNoRef;
-		
-		auto old_service = dynamic_cast<ServiceNoRef *>(m_service.get());
+		auto old_service = dynamic_cast<ServiceT *>(m_service.get());
 		// if the same service and arguments were used, leave old info
-		if (old_service != nullptr && *old_service == service)
+		if (old_service != nullptr && *old_service == *service)
 			return;
-		
-		m_service = std::make_shared<ServiceNoRef>(std::forward<ServiceT>(service));
-		m_worker = boost::async(boost::launch::async, boost::bind(&LastFm::Service::fetch, m_service.get()));
-		
+
+		m_service = std::shared_ptr<ServiceT>(service);
+		m_worker = boost::async(boost::launch::async, std::bind(&LastFm::Service::fetch, m_service));
+
 		w.clear();
 		w << "Fetching information...";
 		w.flush();
